### PR TITLE
Add `std::os::fortanix_sgx` module

### PR DESCRIPTION
**Pull request by jethrogb**
This PR adds the `std::os::sgx` module to expose platform-specific APIs behind the `sgx_platform` feature gate.

Depends on https://github.com/rust-lang/rust/pull/56972 to be able to meaningfully build `std::os` documentation for non-standard targets.

Tracking issue: https://github.com/rust-lang/rust/issues/56975

### CODE REVIEW

File: src/libstd/Cargo.toml [hunk 1]
```diff
@@ -40,7 +40,7 @@ rustc_tsan = { path = "../librustc_tsan" }
 dlmalloc = { version = "0.1", features = ['rustc-dep-of-std'] }
 
 [target.x86_64-fortanix-unknown-sgx.dependencies]
-fortanix-sgx-abi = { version = "0.3.1", features = ['rustc-dep-of-std'] }
+fortanix-sgx-abi = { version = "0.3.2", features = ['rustc-dep-of-std'] }
```
Comment (jethrogb) [inline src/libstd/Cargo.toml]: Diff: https://github.com/fortanix/rust-sgx/commit/769d8062ec404c6a5d7f1dbe69e0d9c704335d16 (adding stability attributes)

File: src/libstd/os/mod.rs [hunk 2]
```diff
@@ -63,4 +48,19 @@ cfg_if! {
     }
 }
 
+#[cfg(target_os = "android")]    pub mod android;
```
Comment (jethrogb) [inline src/libstd/os/mod.rs]: These changes are from #56972

**Comment by joshtriplett** [comment]
cc @jsakkine-intel

**Comment by joshtriplett** [comment]
I don't think it's reasonable for `std::os::sgx` to be an interface to specifically the *fortanix* SGX ABI, as opposed to some other SGX ABI. SGX enclaves can define any number of ABIs.

I would suggest calling this either `std::os::sgx_fortanix` or `std::os::fortanix`.

**Comment by jethrogb** [comment]
Ok, changed to `std::os::fortanix_sgx`.

**Comment by joshtriplett** [comment]
Thanks!

@bors r+

**Pull request merged**